src/nvim/path.c

```diff
@@ -1098,7 +1102,7 @@ int gen_expand_wildcards(int num_pat, char_u **pat, int *num_file,
     p = pat[i];
 
     if (vim_backtick(p))
-      add_pat = expand_backtick(&ga, p, flags);
+      add_pat = (int)expand_backtick(&ga, p, flags);
```

Comment:
Hi. Yes, In new/refactored code, all casts should be safe, which means asserting value is in bounds before casting. In this case, you know `expand_backtick`was previously returning an int, but from the very moment you change its return type into a `size_t`, you cannot rely on that anymore.